### ISSUE TITLE
How can we extract the scalar_mix_0 parameters from an ELMO model?

### ISSUE THREAD

**Issue by SureshArr** [allenai/allennlp#2566]
- OS
 - Python version: 3.6.8
 - AllenNLP version: v0.8.2
 - PyTorch version: 0.6.1

For me elmo.scalar_mix_0.gamma = 1.
and, elmo.scalar_mix_0.scalar_parameters[0].detach() = 0. (same for [1] and [2])

**Comment by nelson-liu** [comment]
Sorry if obvious, but are you doing this on a freshly-loaded `Elmo` instance? The scalar parameters are initialized to those values.

**Comment by SureshArr** [comment]
Thanks for the prompt response!  And sorry if I am understanding this wrong.
Since the ELMO instance is initialized using a weights file, I thought that the weights would be the optimized used for outputting the weighted token embeddings.  And, I queried these weights after I ran a few sentences through the model - these weights don't change.
By the way, I am only trying to obtain token embeddings to plug into another model (i.e., trusting ELMO to provide me pre-trained context sensitive embeddings), not to train the ELMO model itself.  Do these weights change only upon retraining the model? Thanks!

**Comment by nelson-liu** [comment]
> I just want to know the optimum scalar weights used by the model.

The optimum scalar weights are trained on the end task (e.g., NER, QA, etc). By default, the scalar weights are initialized to the same value (so you essentially get an average of the 3 layers). Through gradient descent, the scalar weights are updated to optimize for whatever end task the model is being used for. Indeed, it seems like the scalar weights vary quite a bit depending on the nature of the end task.

Hopefully that clarifies your question? I'm closing this for now, but feel free to reopen if you have more questions.
